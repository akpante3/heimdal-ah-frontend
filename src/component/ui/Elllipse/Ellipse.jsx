/* eslint-disable react/jsx-no-bind */
/* eslint-disable require-jsdoc */
import React, { Component } from 'react';
import FontAwesome from 'react-fontawesome';
import { Dropdown, DropdownToggle, DropdownMenu } from 'reactstrap';
import ModalButton from '../Modal/Index';
import CommentEditHistory from '../CommentEditHistory/CommentEditHistory';

class Ellipse extends Component {
  state = {
    dropDown: false,
  };

  toggle = () => {
    this.setState(prevState => ({
      dropDown: !prevState.dropDown,
    }));
  };

  render() {
    const { dropDown } = this.state;
    const { articleId, commentId } = this.props;

    return (
      <Dropdown isOpen={dropDown} toggle={this.toggle} tag='span'>
        <DropdownToggle
          tag='span'
          onClick={this.toggle}
          data-toggle='dropdown'
          aria-expanded={dropDown}
          className='small_text ellipse'
        >
          <FontAwesome
            name='menu'
            className='fas fa-ellipsis-v'
            onClick={this.toggle}
          />
        </DropdownToggle>
        <DropdownMenu>
          <ModalButton type='signup' label='view history' title='Edit History'>
            <CommentEditHistory articleId={articleId} commentId={commentId} />
          </ModalButton>
        </DropdownMenu>
      </Dropdown>
    );
  }
}

export default Ellipse;
