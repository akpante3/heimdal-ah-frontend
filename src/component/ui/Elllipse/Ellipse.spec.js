/* eslint-disable no-undef */
/* eslint-disable react/jsx-filename-extension */
/* eslint-disable react/react-in-jsx-scope */
import Ellipse from './Ellipse';

describe('The MainError Component Test Suite', () => {
  it('should render the MainError component', () => {
    const wrapper = shallow(<Ellipse />);
    expect(wrapper).toMatchSnapshot();
  });
});
