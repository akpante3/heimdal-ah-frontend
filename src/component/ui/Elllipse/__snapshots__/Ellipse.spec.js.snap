// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`The MainError Component Test Suite should render the MainError component 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Ellipse />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "active": false,
      "addonType": false,
      "children": Array [
        <DropdownToggle
          aria-expanded={false}
          aria-haspopup={true}
          className="small_text ellipse"
          color="secondary"
          data-toggle="dropdown"
          onClick={[Function]}
          tag="span"
        >
          <FontAwesome
            className="fas fa-ellipsis-v"
            name="menu"
            onClick={[Function]}
          />
        </DropdownToggle>,
        <DropdownMenu
          flip={true}
          tag="div"
        >
          <TriggerModal
            label="view history"
            title="Edit History"
            type="signup"
          >
            <Connect(CommentEditHistory) />
          </TriggerModal>
        </DropdownMenu>,
      ],
      "direction": "down",
      "inNavbar": false,
      "isOpen": false,
      "nav": false,
      "setActiveFromChild": false,
      "tag": "span",
      "toggle": [Function],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "aria-expanded": false,
          "aria-haspopup": true,
          "children": <FontAwesome
            className="fas fa-ellipsis-v"
            name="menu"
            onClick={[Function]}
          />,
          "className": "small_text ellipse",
          "color": "secondary",
          "data-toggle": "dropdown",
          "onClick": [Function],
          "tag": "span",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "className": "fas fa-ellipsis-v",
            "name": "menu",
            "onClick": [Function],
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "children": <TriggerModal
            label="view history"
            title="Edit History"
            type="signup"
          >
            <Connect(CommentEditHistory) />
          </TriggerModal>,
          "flip": true,
          "tag": "div",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": <Connect(CommentEditHistory) />,
            "label": "view history",
            "title": "Edit History",
            "type": "signup",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "articleId": undefined,
              "commentId": undefined,
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          "type": [Function],
        },
        "type": [Function],
      },
    ],
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "active": false,
        "addonType": false,
        "children": Array [
          <DropdownToggle
            aria-expanded={false}
            aria-haspopup={true}
            className="small_text ellipse"
            color="secondary"
            data-toggle="dropdown"
            onClick={[Function]}
            tag="span"
          >
            <FontAwesome
              className="fas fa-ellipsis-v"
              name="menu"
              onClick={[Function]}
            />
          </DropdownToggle>,
          <DropdownMenu
            flip={true}
            tag="div"
          >
            <TriggerModal
              label="view history"
              title="Edit History"
              type="signup"
            >
              <Connect(CommentEditHistory) />
            </TriggerModal>
          </DropdownMenu>,
        ],
        "direction": "down",
        "inNavbar": false,
        "isOpen": false,
        "nav": false,
        "setActiveFromChild": false,
        "tag": "span",
        "toggle": [Function],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "aria-expanded": false,
            "aria-haspopup": true,
            "children": <FontAwesome
              className="fas fa-ellipsis-v"
              name="menu"
              onClick={[Function]}
            />,
            "className": "small_text ellipse",
            "color": "secondary",
            "data-toggle": "dropdown",
            "onClick": [Function],
            "tag": "span",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "className": "fas fa-ellipsis-v",
              "name": "menu",
              "onClick": [Function],
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "children": <TriggerModal
              label="view history"
              title="Edit History"
              type="signup"
            >
              <Connect(CommentEditHistory) />
            </TriggerModal>,
            "flip": true,
            "tag": "div",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": <Connect(CommentEditHistory) />,
              "label": "view history",
              "title": "Edit History",
              "type": "signup",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "articleId": undefined,
                "commentId": undefined,
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            "type": [Function],
          },
          "type": [Function],
        },
      ],
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
